packages/messaging/index.js

```diff
@@ -36,12 +36,22 @@ export class MessagingContext {
      * @param {string} params.context
      * @param {string} params.featureName
      * @param {"production" | "development"} params.env
+     * @param {((name: string, data?: Record<string, any>) => string) | null | undefined} [params.createId]
+     *   Optional callback for overriding how message IDs are generated.
      * @internal
      */
     constructor (params) {
         this.context = params.context
         this.featureName = params.featureName
         this.env = params.env
+
+        /**
+         * This allows for deterministic tests in all environments. For example,
+         * in a test situation you can ensure the ID is known.
+         */
+        this.createId = params.createId || ((name) => {
+            return globalThis?.crypto?.randomUUID?.() || name + '.response'
+        })
```

Comment:
This behaves exactly as before, but also now acts as DI for overriding the side-effecting nature of `globalThis?.crypto?.randomUUID`